src/celestia/viewmanager.cpp

```diff
@@ -0,0 +1,342 @@
+}
+
+
+void
+ViewManager::tryStartResizing(const WindowMetrics& metrics, float x, float y)
+{
+    View* v1 = nullptr;
+    View* v2 = nullptr;
+    for (View* v : m_views)
+    {
+        if (v->type != View::ViewWindow)
+            continue;
+
+        auto [vx, vy, vxp, vyp] = viewBoundaryTestValues(v, metrics, x, y);
+        if ((vx >=0 && vx <= 1 && (std::abs(vyp) <= 2 || std::abs(vyp - v->height * metrics.height) <= 2)) ||
+            (vy >=0 && vy <= 1 && (std::abs(vxp) <= 2 || std::abs(vxp - v->width * metrics.width) <= 2)))
+        {
+            if (v1 == nullptr)
+            {
+                v1 = v;
+            }
+            else
+            {
+                v2 = v;
+                break;
+            }
+        }
+    }
+
+    if (v2 == nullptr)
+        return;
+
+    // Look for common ancestor to v1 & v2 = split being draged.
+    View *p1 = v1;
+    View *p2 = v2;
+    while ((p1 = p1->parent) != nullptr )
+    {
+        p2 = v2;
+        while (((p2 = p2->parent) != nullptr) && p1 != p2) ;
+        if (p2 != nullptr)
+            break;
+    }
+
+    if (p2 != nullptr)
+        m_resizeSplit = p1;
+}
+
+
+bool
+ViewManager::resizeViews(const WindowMetrics& metrics, float dx, float dy)
+{
+    if (!isResizing())
+        return false;
+
+    switch(m_resizeSplit->type)
+    {
+    case View::HorizontalSplit:
+        if (m_resizeSplit->walkTreeResizeDelta(m_resizeSplit->child1, dy / metrics.height, true) &&
+            m_resizeSplit->walkTreeResizeDelta(m_resizeSplit->child2, dy / metrics.height, true))
+        {
+            m_resizeSplit->walkTreeResizeDelta(m_resizeSplit->child1, dy / metrics.height, false);
+            m_resizeSplit->walkTreeResizeDelta(m_resizeSplit->child2, dy / metrics.height, false);
+        }
+        break;
+    case View::VerticalSplit:
+        if (m_resizeSplit->walkTreeResizeDelta(m_resizeSplit->child1, dx / metrics.width, true) &&
+            m_resizeSplit->walkTreeResizeDelta(m_resizeSplit->child2, dx / metrics.width, true))
+        {
+            m_resizeSplit->walkTreeResizeDelta(m_resizeSplit->child1, dx / metrics.width, false);
+            m_resizeSplit->walkTreeResizeDelta(m_resizeSplit->child2, dx / metrics.width, false);
+        }
+        break;
+    case View::ViewWindow:
+        break;
+    }
+
+    return true;
+}
+
+
+bool
+ViewManager::stopResizing()
+{
+    if (m_resizeSplit == nullptr)
+        return false;
+
+    m_resizeSplit = nullptr;
+    return true;
+}
+
+
+ViewSplitResult
+ViewManager::splitView(Simulation* sim, View::Type type, View* av, float splitPos)
+{
+    if (type == View::ViewWindow)
+        return ViewSplitResult::Ignored;
+
+    if (av == nullptr)
+        av = *m_activeView;
+
+    if (!av->isSplittable(type))
+        return ViewSplitResult::NotSplittable;
+
+    Observer* o = sim->addObserver();
+
+    // Make the new observer a copy of the old one
+    // TODO: This works, but an assignment operator for Observer
```

Comment:
could you address this as well?

Reply:
We do have copy constructor/assignment operators, so probably this comment is just outdated? Will remove it.

Reply:
made the method duplicate the active observer, since nothing else was calling it.